crates/nu-cli/src/completions/variable_completions.rs

```diff
@@ -131,126 +44,128 @@ impl Completer for VariableCompletion {
             });
         }
 
+        let mut add_candidate = |name, var_id: &VarId| {
+            matcher.add_semantic_suggestion(SemanticSuggestion {
+                suggestion: Suggestion {
+                    value: String::from_utf8_lossy(name).to_string(),
+                    span: current_span,
+                    ..Suggestion::default()
+                },
+                kind: Some(SuggestionKind::Type(
+                    working_set.get_variable(*var_id).ty.clone(),
+                )),
+            })
+        };
+
         // TODO: The following can be refactored (see find_commands_by_predicate() used in
         // command_completions).
         let mut removed_overlays = vec![];
         // Working set scope vars
         for scope_frame in working_set.delta.scope.iter().rev() {
             for overlay_frame in scope_frame.active_overlays(&mut removed_overlays).rev() {
-                for v in &overlay_frame.vars {
-                    matcher.add_semantic_suggestion(SemanticSuggestion {
-                        suggestion: Suggestion {
-                            value: String::from_utf8_lossy(v.0).to_string(),
-                            span: current_span,
-                            ..Suggestion::default()
-                        },
-                        kind: Some(SuggestionKind::Type(
-                            working_set.get_variable(*v.1).ty.clone(),
-                        )),
-                    });
+                for (name, var_id) in &overlay_frame.vars {
+                    add_candidate(name, var_id);
                 }
             }
         }
-
         // Permanent state vars
         // for scope in &self.engine_state.scope {
         for overlay_frame in working_set
             .permanent_state
             .active_overlays(&removed_overlays)
             .rev()
         {
-            for v in &overlay_frame.vars {
-                matcher.add_semantic_suggestion(SemanticSuggestion {
-                    suggestion: Suggestion {
-                        value: String::from_utf8_lossy(v.0).to_string(),
-                        span: current_span,
-                        ..Suggestion::default()
-                    },
-                    kind: Some(SuggestionKind::Type(
-                        working_set.get_variable(*v.1).ty.clone(),
-                    )),
-                });
+            for (name, var_id) in &overlay_frame.vars {
+                add_candidate(name, var_id);
             }
         }
 
         matcher.results()
     }
 }
 
-// Find recursively the values for sublevels
-// if no sublevels are set it returns the current value
-fn nested_suggestions(
-    val: &Value,
-    sublevels: &[Vec<u8>],
-    current_span: reedline::Span,
-) -> Vec<SemanticSuggestion> {
-    let mut output: Vec<SemanticSuggestion> = vec![];
-    let value = recursive_value(val, sublevels).unwrap_or_else(Value::nothing);
+pub struct CellPathCompletion<'a> {
```

Comment:
I know this used to be part of the variable completer, but I'd prefer to have this in a separate file, so we can continue the one-completer-per-file thing we have going right now.